packages/web-components/src/components/link-with-icon/link-with-icon.ts

```diff
@@ -29,7 +31,10 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
  * @slot icon-left - The CTA icon to place at the left.
  */
 @customElement(`${ddsPrefix}-link-with-icon`)
-class DDSLinkWithIcon extends StableSelectorMixin(BXLink) {
+class DDSLinkWithIcon extends CTAMixin(StableSelectorMixin(BXLink)) {
```

Comment:
I've added the `CTAMixin` here so that you could set `cta-type` on a `dds-link-with-icon` or it's derrivatives and have the icon rtl logic decide on the right arrow. I _think_ that's OK based on the fact that this [change was made in v2](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/web-components/src/components/link-with-icon/link-with-icon.ts#L34), but open to thoughts. Alternatively, we could leave it and let the onus be on the client to slot in the right icon, but that's a real pain.

Reply:
I'm in favor of this change. Makes me wonder if we should give `<dds-card-footer>` the same treatment.